src/state/routing/gas.ts

```diff
@@ -26,7 +26,7 @@ export async function getApproveInfo(
   // If any of these arguments aren't provided, then we cannot generate approval cost info
   if (!account || !usdCostPerGas) return { needsApprove: false }
 
-  const provider = RPC_PROVIDERS[currency.chainId as SupportedInterfaceChain]
+  const provider = DEPRECATED_RPC_PROVIDERS[currency.chainId as SupportedInterfaceChain]
```

Comment:
why aren't we switching between these and the non-deprecated based on the feature flag here?

Reply:
because getApproveInfo is not a hook, and the current feature gate api is hook-only. Since this call site will be called very rarely compared to others, I chose to use the deprecated version instead and make the change to the new one when we remove the gate. I can see a reasonable argument for doing the opposite, though!